### PR TITLE
plat-sam: registers sama7g5 clocks for SCMI usage

### PR DESCRIPTION
**Pull request by TonyHan11**

### CODE REVIEW

**Comment by TonyHan11** [comment]
@tprrt commit [272c691](https://github.com/OP-TEE/optee_os/pull/6721/commits/272c691baa97bd409677121241df40058f54cd67) [179d59d](https://github.com/OP-TEE/optee_os/pull/6721/commits/179d59d7bb33dc6e5710ede93b42c1f0f6f9802c) are cherry-picked from [6e036ad](https://github.com/tprrt/optee_os/commit/6e036ada252a4f57c37a4d5c31a0d0230444e965) [c8bbbe0](https://github.com/tprrt/optee_os/commit/c8bbbe0375594c0db8660b78b779a2a2c13960f5) in [sama7g54_optee](https://github.com/tprrt/optee_os/commits/tprrt/sama7g54_optee/) branch with conflicts fixed.

**Comment by tprrt** [comment]
> @tprrt commit [272c691](https://github.com/OP-TEE/optee_os/pull/6721/commits/272c691baa97bd409677121241df40058f54cd67) [179d59d](https://github.com/OP-TEE/optee_os/pull/6721/commits/179d59d7bb33dc6e5710ede93b42c1f0f6f9802c) are cherry-picked from [6e036ad](https://github.com/tprrt/optee_os/commit/6e036ada252a4f57c37a4d5c31a0d0230444e965) [c8bbbe0](https://github.com/tprrt/optee_os/commit/c8bbbe0375594c0db8660b78b779a2a2c13960f5) in [sama7g54_optee](https://github.com/tprrt/optee_os/commits/tprrt/sama7g54_optee/) branch with conflicts fixed.

Thanks @TonyHan11

File: core/arch/arm/plat-sam/freq.c [hunk 1]
```diff
@@ -22,7 +22,11 @@ static TEE_Result get_freq_from_dt(void)
 	if (!fdt)
 		panic();
 
+#ifdef CFG_SAMA7G5
+	node = fdt_node_offset_by_compatible(fdt, -1, "arm,cortex-a7");
+#else
 	node = fdt_node_offset_by_compatible(fdt, -1, "arm,cortex-a5");
+#endif
```
Comment (jforissier) [inline core/arch/arm/plat-sam/freq.c:29]: Prefer:
```C
	if (IS_ENABLED(CFG_SAMA7G5))
		node = fdt_node_offset_by_compatible(fdt, -1, "arm,cortex-a7");
	else
		node = fdt_node_offset_by_compatible(fdt, -1, "arm,cortex-a5");
```

**Review by jforissier — COMMENTED** [review]
Minor comments:
- "plat-sam: register..." instead of "plat-sam: registers..."
- s/each SCMI clocks/each SCMI clock/
- one more below

With that fixed:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by TonyHan11** [comment]
@jforissier  Updated according to the comments and added the review tags, thanks.

**Pull request merged**